bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/hbasesnapshots/EndToEndIT.java

```diff
@@ -0,0 +1,191 @@
+ * -Dgoogle.bigtable.project.id=<project_id> \
+ * -Dgoogle.bigtable.instance.id=<instance_id> \
+ * -Dgoogle.dataflow.stagingLocation=gs://<test_bucket>/staging \
+ * -Dcloud.test.data.folder=gs://<test_bucket>/integration-test/
+ */
+public class EndToEndIT {
+
+  private static final String TEST_SNAPSHOT_NAME = "test-snapshot";
+  // Location of test data hosted on Google Cloud Storage, for on-cloud dataflow tests.
+  private static final String CLOUD_TEST_DATA_FOLDER = "cloud.test.data.folder";
+
+  // Column family name used in all test bigtables.
+  private static final String CF = "cf";
+
+  // Full path of the Cloud Storage folder where dataflow jars are uploaded to.
+  private static final String GOOGLE_DATAFLOW_STAGING_LOCATION = "google.dataflow.stagingLocation";
+
+  private Connection connection;
+  private String projectId;
+  private String instanceId;
+  private String tableId;
+
+  private GcsUtil gcsUtil;
+  private String dataflowStagingLocation;
+  private String workDir;
+  private byte[][] keySplits;
+
+  // Snapshot data setup
+  private String hbaseSnapshotDir;
+  private String restoreDir;
+
+  @Before
+  public void setup() throws Exception {
+    projectId = getTestProperty(BigtableOptionsFactory.PROJECT_ID_KEY);
+    instanceId = getTestProperty(BigtableOptionsFactory.INSTANCE_ID_KEY);
+    dataflowStagingLocation = getTestProperty(GOOGLE_DATAFLOW_STAGING_LOCATION);
+    String cloudTestDataFolder = getTestProperty(CLOUD_TEST_DATA_FOLDER);
+    if (!cloudTestDataFolder.endsWith(File.separator)) {
+      cloudTestDataFolder = cloudTestDataFolder + File.separator;
+    }
+
+    hbaseSnapshotDir = cloudTestDataFolder + "data/";
+    UUID test_uuid = UUID.randomUUID();
+    restoreDir = cloudTestDataFolder + "restore/" + test_uuid;
+
+    // Cloud Storage config
+    GcpOptions gcpOptions = PipelineOptionsFactory.create().as(GcpOptions.class);
+    gcpOptions.setProject(projectId);
+    gcsUtil = new GcsUtil.GcsUtilFactory().create(gcpOptions);
+
+    // Bigtable config
+    connection = BigtableConfiguration.connect(projectId, instanceId);
+    tableId = "test_" + UUID.randomUUID().toString();
+
+    System.out.println("Setting up integration tests");
+
+    String[] keys = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
+    keySplits = new byte[keys.length][];
+    for (int i = 0; i < keys.length; i++) {
+      keySplits[i] = keys[i].getBytes();
+    }
+  }
+
+  private static String getTestProperty(String name) {
+    return checkNotNull(System.getProperty(name), "Required property missing: " + name);
+  }
+
+  @After
+  public void teardown() throws IOException {
+    final List<GcsPath> paths = gcsUtil.expand(GcsPath.fromUri(restoreDir + "/*"));
+
+    if (!paths.isEmpty()) {
+      final List<String> pathStrs = new ArrayList<>();
+
+      for (GcsPath path : paths) {
+        pathStrs.add(path.toString());
+      }
+      this.gcsUtil.remove(pathStrs);
```

Comment:
Can't we just delete the whole restore folder?

Reply:
This is following the prior practice in the sequencefiles import tests, the reason being we didn't create the folder in the setup, so it should not be deleted during teardown.

If we use setup to create this folder, then the setup and teardown will also need to deal with more cases like:
1, if the folder already exists?
2. the folder created but the setup it not successful?
3. the folder didn't got created so the teardown will be deleting an non-existing folder.

So for simplicity and conformance with the prior practice, I would suggest that we keep current practice.